cli/tests/activate.bats

```diff
@@ -2081,3 +2081,84 @@ EOF
 }
 
 # ---------------------------------------------------------------------------- #
+
+# bats test_tags=activate,activate:nested_flox_activate_tracelevel
+@test "{bash,fish,tcsh,zsh}: confirm _flox_activate_tracelevel set in nested activation" {
+  project_setup
+
+  # The shell-specific flox init scripts finish by unsetting the
+  # _flox_activate_tracelevel environment variable, and this can
+  # cause problems for an "outer" interactive activation when there
+  # is an "inner" in-place activation happening by way of a "dotfile".
+
+  # Set up this test by creating dotfiles which perform an in-place
+  # activation, and then run an interactive activation of a second
+  # environment to confirm that _flox_activate_tracelevel is set
+  # for the outer activation.
+
+  # Start by adding logic to create semaphore files for all shells.
+  for i in "$HOME/.bashrc.extra" "$HOME/.config/fish/config.fish.extra" "$HOME/.tcshrc.extra" "$HOME/.zshrc.extra"; do
+    cat > "$i" <<EOF
+touch "$PROJECT_DIR/_flox_activate_tracelevel.in_test"
+test -n "\$_flox_activate_tracelevel" || touch "$PROJECT_DIR/_flox_activate_tracelevel.not_defined"
+EOF
+  done
+
+  # Finish by appending shell-specific flox activation syntax.
+  for i in "$HOME/.bashrc.extra" "$HOME/.config/fish/config.fish.extra" "$HOME/.zshrc.extra"; do
+    echo "eval \"\$($FLOX_BIN activate --dir $PROJECT_DIR)\"" >> "$i"
+  done
+  echo "eval \"\`$FLOX_BIN activate --dir $PROJECT_DIR\`\"" >> "$HOME/.tcshrc.extra"
```

Comment:
Non-blocking, curious: is this separate because `tcsh` doesn't support `$()`?

Reply:
Correct - `tcsh` does not support `$()`, but in addition `bash` and `zsh` _do_ support backticks (but we haven't been testing that). For the avoidance of doubt `fish` does _not_ support backticks.

So we could actually have turned this upside-down and made `fish` the exception!